File: ios/BiangBiang Hanzi/AppDesign.swift
//
//  AppDesign.swift
//  BiangBiang Hanzi
//
//  Shared design constants used across the app for a consistent look.
//

import Foundation

enum AppDesign {
    /// Default corner radius for cards and rounded containers.
    static let cornerRadius: CGFloat = 12
    /// Larger corner radius for prominent containers (toasts, banners).
    static let cornerRadiusLarge: CGFloat = 16
    /// Compact corner radius for small surfaces (overlay labels).
    static let cornerRadiusCompact: CGFloat = 6

    /// Spacing between sections within a screen.
    static let sectionSpacing: CGFloat = 20
    /// Standard horizontal padding for content edges.
    static let horizontalPadding: CGFloat = 20
    /// Padding above the bottom safe area for floating controls.
    static let bottomToolbarPadding: CGFloat = 40
    /// Inner spacing for stacks of related controls.
    static let stackSpacing: CGFloat = 12

    /// Standard tap target size (HIG minimum).
    static let tapTarget: CGFloat = 44

    /// Default duration for short interaction animations.
    static let shortAnimation: Double = 0.2
    /// Toast fade duration.
    static let toastAnimation: Double = 0.25

    /// Standard shadow radius for floating buttons.
    static let buttonShadow: CGFloat = 4
}

File: android/app/src/main/java/dev/veeso/harakatlens/ui/AppDesign.kt
package dev.veeso.harakatlens.ui

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp

/**
 * Shared design constants used across the app for a consistent look.
 */
object AppDesign {
    /** Default corner radius for cards and rounded containers. */
    val cornerRadius = 12.dp

    /** Larger corner radius for prominent containers (toasts, banners). */
    val cornerRadiusLarge = 16.dp

    /** Compact corner radius for small surfaces (overlay labels). */
    val cornerRadiusCompact = 6.dp

    /** Spacing between sections within a screen. */
    val sectionSpacing = 20.dp

    /** Standard horizontal padding for content edges. */
    val horizontalPadding = 20.dp

    /** Padding above the bottom safe area for floating controls. */
    val bottomToolbarPadding = 40.dp

    /** Inner spacing for stacks of related controls. */
    val stackSpacing = 12.dp

    /** Standard tap target size. */
    val tapTarget = 44.dp

    /** Shutter button size. */
    val shutterSize = 72.dp

    /** Default duration for short interaction animations. */
    const val SHORT_ANIMATION_MS = 200

    /** Toast fade duration. */
    const val TOAST_ANIMATION_MS = 250

    /** Debounce window for live text input processing. */
    const val INPUT_DEBOUNCE_MS = 800L

    /** Brand color (Harakat green). */
    val brandGreen = Color(0xFF006C35)

    /** Lighter brand variant (HSL L=30%). */
    val brandGreenLight = Color(0xFF00994C)

    /** Darker brand variant (HSL L=15%). */
    val brandGreenDark = Color(0xFF004C26)
}
